src/test/java/io/camunda/zeebe/bpmnassert/testengine/db/InMemoryZeebeDbTransactionTest.java

```diff
@@ -0,0 +1,479 @@
+        });
+  }
+
+  @Test
+  public void shouldStartNewTransaction() throws Exception {
+    // given
+    oneKey.wrapLong(1);
+    oneValue.wrapLong(-1);
+
+    twoKey.wrapLong(52000);
+    twoValue.wrapLong(192313);
+
+    threeKey.wrapLong(Short.MAX_VALUE);
+    threeValue.wrapLong(Integer.MAX_VALUE);
+
+    final ZeebeDbTransaction transaction = transactionContext.getCurrentTransaction();
+    transaction.run(
+        () -> {
+          oneColumnFamily.put(oneKey, oneValue);
+          twoColumnFamily.put(twoKey, twoValue);
+          threeColumnFamily.put(threeKey, threeValue);
+        });
+
+    // when
+    transaction.commit();
+
+    // then
+    assertThat(oneColumnFamily.exists(oneKey)).isTrue();
+    assertThat(twoColumnFamily.exists(twoKey)).isTrue();
+    assertThat(threeColumnFamily.exists(threeKey)).isTrue();
+  }
+
+  @Test
+  public void shouldAccessOnOpenTransaction() throws Exception {
+    // given
+    oneKey.wrapLong(1);
+    oneValue.wrapLong(-1);
+
+    twoKey.wrapLong(52000);
+    twoValue.wrapLong(192313);
+
+    threeKey.wrapLong(Short.MAX_VALUE);
+    threeValue.wrapLong(Integer.MAX_VALUE);
+
+    final ZeebeDbTransaction transaction = transactionContext.getCurrentTransaction();
+    transaction.run(
+        () -> {
+          oneColumnFamily.put(oneKey, oneValue);
+          twoColumnFamily.put(twoKey, twoValue);
+          threeColumnFamily.put(threeKey, threeValue);
+        });
+
+    // when
+    // no commit
+
+    // then
+    // uses the same transaction
+    assertThat(oneColumnFamily.exists(oneKey)).isTrue();
+    assertThat(twoColumnFamily.exists(twoKey)).isTrue();
+    assertThat(threeColumnFamily.exists(threeKey)).isTrue();
+  }
+
+  @Test
+  public void shouldNotReopenTransaction() throws Exception {
+    // given
+    final ZeebeDbTransaction transaction = transactionContext.getCurrentTransaction();
+
+    transaction.run(
+        () -> {
+
+          // when
+          final ZeebeDbTransaction sameTransaction = transactionContext.getCurrentTransaction();
+
+          // then
+          assertThat(transaction).isEqualTo(sameTransaction);
+        });
+  }
+
+  @Test
+  public void shouldNotReopenTransactionWithOperations() {
+    // given
+    oneKey.wrapLong(1);
+    oneValue.wrapLong(-1);
+
+    twoKey.wrapLong(52000);
+    twoValue.wrapLong(192313);
+
+    threeKey.wrapLong(Short.MAX_VALUE);
+    threeValue.wrapLong(Integer.MAX_VALUE);
+
+    transactionContext.runInTransaction(
+        () -> {
+
+          // when
+          final ZeebeDbTransaction sameTransaction = transactionContext.getCurrentTransaction();
+          sameTransaction.run(
+              () -> {
+                oneColumnFamily.put(oneKey, oneValue);
+                twoColumnFamily.put(twoKey, twoValue);
+                threeColumnFamily.put(threeKey, threeValue);
+              });
+          sameTransaction.commit();
+
+          // then it is committed but available in this transaction
```

Comment:
❓ I'm not sure about this test. I feel like this makes it seem like `sameTransaction` is still active. However, this is not the case. With the `commit()` this transaction was ended. 2 lines down from this one a new transaction gets started with `oneColumnFamily.delete(oneKey)`.

I don't know what the intended behaviour of this test is. The test says it should not reopen a transaction, but a new transaction does get started. So I'm a little bit confused here.

Reply:
Pretty much just copy and pasted it from EZE, tbh.